### PR TITLE
Handle batch norm in target update

### PR DESCRIPTION
**Pull request by honglu2875**
Change target net update behavior by further copying batch norm stats. See https://github.com/DLR-RM/stable-baselines3/issues/996

## Description
In QRDQN and TQC, batch norm stats were not touched in target net update as they are not included in `Module.parameters()`. Now they are directly copied in target net update regardless of `tau`.

## Context
https://github.com/DLR-RM/stable-baselines3/issues/996
(Let me know if I also need to raise an issue in this sb3-contrib repo.)
- [x] I have raised an issue to propose this change ([required](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/blob/master/CONTRIBUTING.md))

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] The functionality/performance matches that of the source (**required** for new training algorithms or training-related features).
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have included an example of using the feature (*required for new features*).
- [ ] I have included baseline results (**required** for new training algorithms or training-related features).
- [ ] I have updated the documentation accordingly.
- [x] I have updated the changelog accordingly (**required**).
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: tests/test_cnn.py [hunk 1]
```diff
@@ -109,14 +109,21 @@ def test_feature_extractor_target_net(model_class, share_features_extractor):
...
 
     # Critic and target should differ
     params_should_differ(model.critic.parameters(), model.critic_target.parameters())
 
+    # Force stats copy
+    model.target_update_interval = 1
+    model._on_step()
+
+    # running stats are copied, see https://github.com/DLR-RM/stable-baselines3/issues/996
+    params_should_match(model.batch_norm_stats, model.batch_norm_stats_target)
```
Comment (araffin) [inline tests/test_cnn.py:125]: `model.batch_norm_stats` is an empty list here, no?

**Review by araffin — COMMENTED** [review]
LGTM, just one minor comment.
Btw, did you had time to try the fix on your task? (meaning does batch norm works now?)

File: tests/test_cnn.py (hunk 1 above)
Comment (honglu2875) [reply tests/test_cnn.py:125]: Oh I was thinking since this is after line 115, I'd better force the target copy and try to check. Why is it an empty set?

**Comment by honglu2875** [comment]
> LGTM, just one minor comment. Btw, did you had time to try the fix on your task? (meaning does batch norm works now?)

Yes. Just did a small test on the same set of parameters (target updated every 1000 steps instead of 10000). Orange is with the older version and black is our latest. Looks like it improves the convergence of loss.

Screenshot from 2022-08-26 17-27-02

Loss can still improve in some cases though (like the one below, orange is the old one). But the actual performance may still be bad.

Screenshot from 2022-08-26 17-24-45

These are all on my custom environment though. I tried to test it on those standard gym environment today but for some reason I couldn't get it up and running (errors like numpy array is not hashable).

**Review by araffin — APPROVED** [review]
LGTM, thanks =)

**Comment by Nelson-iitp** [comment]
can I do ```requires_grad=False``` on target network parameters and always keep it in ```eval``` mode?
